src/core/lib/gprpp/work_serializer.cc

```diff
@@ -102,10 +112,14 @@ void WorkSerializer::WorkSerializerImpl::Run(std::function<void()> callback,
   GPR_DEBUG_ASSERT(GetSize(prev_ref_pair) > 0);
   if (GetOwners(prev_ref_pair) == 0) {
     // We took ownership of the WorkSerializer. Invoke callback and drain queue.
+#ifndef NDEBUG
+    current_thread_ = std::this_thread::get_id();
+#endif
     if (GRPC_TRACE_FLAG_ENABLED(grpc_work_serializer_trace)) {
       gpr_log(GPR_INFO, "  Executing immediately");
     }
     callback();
+    callback = nullptr;  // Delete while still holding WorkSerializer.
```

Comment:
ah I see! Can you expand the comment to say something like - The callback might be holding refs that need to be given up from within the WorkSerializer

Reply:
Done.